scipy/sparse/linalg/_isolve/symmlq.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['symmlq']
+
+
+def symmlq(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None,
+           atol=None, haptol=1e-18, show=False):
+    """
+    Use Symmetric LQ iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, ndarray, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, `A` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        `scipy.sparse.linalg.LinearOperator`.
+    b : {ndarray}
+        Right hand side of the linear system. Has shape (N,) or (N, 1).
+    x0 : {ndarray}
+        Starting guess for the solution.
+    tol, atol : float, optional
+        Tolerances for convergence, ``norm(b - Ax) < max(tol*norm(b), atol)``.
+        The default for `tol` is 1.0e-5.
+        The default for `atol` is ``tol * norm(b)``.
+
+        .. warning::
+
+           The default value for `atol` will be changed in a future release.
+           For future compatibility, specify `atol` explicitly.
+    haptol : float, optional
+        Tolerance for happy breakdown, ``abs(np.inner(r.conjugate(), Mr)) <
+        haptol``.
+        Default is 1.0e-18.
+    maxiter : int, optional
+        Maximum number of iterations.  Iteration will stop after maxiter
+        steps even if the specified tolerance has not been achieved.
+        Default is ``min(10000, n * 10)``, where ``n = A.shape[0]``.
+    M : {sparse matrix, ndarray, LinearOperator}
+        Inverse of the preconditioner of A. M should approximate the
+        inverse of A and be easy to solve for (see Notes). Effective
+        preconditioning dramatically improves the rate of convergence,
+        which implies that fewer iterations are needed to reach a given
+        error tolerance. By default, no preconditioner is used.
+    callback : function, optional
+        User-supplied function to call after each iteration. It is called
+        as `callback(rnorm)`, where `rnorm` is the current preconditioned
+        residual norm.
+    show : bool, optional
+        Specify ``show = True`` to show the convergence, ``show = False`` is
+        to close the output of the convergence.
+        Default is `False`.
+
+    Returns
+    -------
+    x : ndarray
+        The converged solution.
+    info : int
```

Comment:
This is a Fortran type behavior because historically they didn't know what the issue was before the error. The function should error out if there is a problem otherwise emit a warning and return the result. Output parameters are extremely difficult to modify in later versions hence we should be very conservative.